src/server.cpp

```diff
@@ -1394,20 +1464,147 @@ void CServer::CreateAndSendChatTextForAllConChannels ( const int      iCurChanID
     QString sCurColor = vstrChatColors[iCurChanID % vstrChatColors.Size()];
 
     const QString strActualMessageText =
-        "<font color=""" + sCurColor + """>(" +
-        QTime::currentTime().toString ( "hh:mm:ss AP" ) + ") <b>" +
+        "<small style='color:" + sCurColor + ";'>[" +
+        QTime::currentTime().toString ( "hh:mm" ) + "] <b>" +
         ChanName.toHtmlEscaped() +
-        "</b></font> " + strChatText;
+        "</b></small><br />" + strChatText;
 
+    // check if chat is disabled
+    if ( bEduModeEnabled && EduModeIsFeatureDisabled( EDU_MODE_FEATURE_CHAT ) )
+    {
+        vecChannels[iCurChanID].CreateChatTextMes( "<span style='color:red;'>ERROR</span>: Chat is disabled." );
+    }
+    else
+    {
+        // Send chat text to all connected clients ---------------------------------
+        for ( int i = 0; i < iMaxNumChannels; i++ )
+        {
+            if ( vecChannels[i].IsConnected() )
+            {
+                // send message
+                vecChannels[i].CreateChatTextMes ( strActualMessageText );
+            }
+        }
+    }
 
-    // Send chat text to all connected clients ---------------------------------
-    for ( int i = 0; i < iMaxNumChannels; i++ )
+}
+void CServer::InterpretAndExecuteChatCommand ( const int iCurChanID,
+                                               const QString& strChatText )
```

Comment:
A function analyzing incoming chat messages and basically enabling the administrative functions